astrokat/utility.py

```diff
@@ -17,6 +17,45 @@ class NoTargetsUpError(Exception):
 def read_yaml(filename):
     with open(filename, 'r') as stream:
         data = yaml.safe_load(stream)
+
+    # handle mapping of user friendly keys to CAM resource keys
+    if 'instrument' in data.keys():
+        instrument = data['instrument']
+        if instrument is not None:
+            if 'integration_period' in instrument.keys():
+                integration_period = float(instrument['integration_period'])
+                instrument['dump_rate'] = 1./integration_period
+                del instrument['integration_period']
+    # verify required information in observation loop before continuing
+    if 'durations' in data.keys():
+        if data['durations'] is None:
+            msg = 'durations primary key cannot be empty in observation YAML file'
+            raise RuntimeError(msg)
+        if 'start_time' in data['durations']:
+            start_time = data['durations']['start_time']
+            if type(start_time) is str:
+                data['durations']['start_time'] = \
+                        datetime.datetime.strptime(start_time, '%Y-%m-%d %H:%M')
+    if 'observation_loop' not in data.keys():
+        raise RuntimeError('Nothing to observer, exiting')
+    if data['observation_loop'] is None:
+        raise RuntimeError('Empty observation loop, exiting')
+    for obs_loop in data['observation_loop']:
+        if type(obs_loop) is str:
+            raise RuntimeError('Expected observation list, got string')
```

Comment:
Error says a list is expected, but the next line uses `obs_loop.keys()`, which suggest it is more like a dict?

Reply:
Yes, the YAML is read into a JSON style dictionary, but this is for the user input.
In the YAML the observation loops are specified as having an LST range + a list of targets.
Thus, giving an error that is not "quite" correct makes the required input a little clearer.

What about extending the error message to something like?
"
RuntimeError("Incomplete observation input: LST range and list of observations required.")
"

Reply:
Suggested implementation
RuntimeError("Incomplete observation input: LST range and a minimum of one target required.")

Reply:
Updated error message implemented for more explicit information to user related to possible error.
+
Added TODO note that needs to be implemented and verified with the implementation of multiple observation loop needed for zoom modes.
Issue #47